### DISCUSSION TITLE
Do not use SetRange with filter expressions

### DISCUSSION THREAD

**Discussion by ChristianHovenbitzer** [StefanMaron/BusinessCentral.LinterCop#32] [New Rule]
Create a warning if a filter expression is used with a SetRange. For instance:

SetRange("Entry No.", '<>0');

Probably a rule which identifies logical operators within SetRange?

**Comment by someC0d3r** [comment]
Hey @ChristianHovenbitzer 

If you set a filter on a field which isn't datatype "Text" you should receive an error:
![image](https://user-images.githubusercontent.com/88825803/135256477-b3ef2123-9181-4c36-92f4-ad61188b8b21.png)


Nevertheless I coded something for text fields - is it going in the correct way?

![image](https://user-images.githubusercontent.com/88825803/135256638-7c1238aa-e8bd-436d-a603-3cd8753c6ba0.png)

It's currently checking on following filter operators:
< > .. * & |

Here is the commit for Rule0008:
https://github.com/someC0d3r/BusinessCentral.LinterCop/commit/de69ef44eeb9fd520b78441baa5bf1639740a0ed

@StefanMaron 
If this rule is okay I would send a pull request over to your repo. :)

Greets
someCoder

**Reply by StefanMaron** [reply]
I added one comment. Rest looking good to me

**Reply by someC0d3r** [reply]
Answered it and pull request created! :)

**Reply by ChristianHovenbitzer** [reply]
Works perfectly! Thanks for the effort
